manta-accounting/src/wallet/signer.rs

```diff
@@ -1111,6 +1175,35 @@ where
         Ok(senders)
     }
 
+    /// Builds two virtual [`Sender`]s for `pre_sender`.
+    #[inline]
+    fn virtual_senders(
+        &mut self,
+        parameters: &Parameters<C>,
+        asset_id: &C::AssetId,
+        pre_sender: PreSender<C>,
+    ) -> Result<[Sender<C>; PrivateTransferShape::SENDERS], SignError<C>>
+    where
+        UtxoAccumulatorModel<C>: Clone,
+    {
+        let mut utxo_accumulator = C::UtxoAccumulator::from(self.utxo_accumulator.model().clone());
+        let sender = pre_sender
+            .insert_and_upgrade(parameters, &mut utxo_accumulator)
+            .expect("Unable to upgrade expected UTXO.");
+        let mut senders = Vec::new();
+        senders.push(sender);
+        let identifier = self.rng.gen();
+        senders.push(
+            self.build_pre_sender(
+                parameters,
+                identifier,
+                Asset::<C>::new(asset_id.clone(), Default::default()),
+            )
+            .upgrade_unchecked(Default::default()),
+        );
```

Comment:
```suggestion
        senders.push(
            self.build_pre_sender(
                parameters,
                self.rng.gen(),
                Asset::<C>::new(asset_id.clone(), Default::default()),
            )
            .upgrade_unchecked(Default::default()),
        );
```

Reply:
Doesn't let me because `build_pre_sender` is already taking a mutable reference of `self`.